### ISSUE TITLE
Suggestion: use websocket instead of polling the status every 1.5 seconds

### ISSUE THREAD

**Issue by xuancong84** [vicwomg/pikaraoke#320]
Hi @vicwomg , just wondering if you are still working on this project, it might be a good idea to use WebSocket (which I was learning recently) to inform each client when the play status has changed. Making an HTTP request every 1.5 seconds per client can be too costly when there are more users.

**Comment by xuancong84** [comment]
> Would definitely like to look into websockets. Hope it's possible.

Actually, I have already implemented WebSocket in my fork, you may want to do a pull and merge, but it will be tough as there are too many changes. I am currently interested and doing adding songs by speaking the song title, using OpenAI's Whisper model to do speech-to-text.

**Comment by killerbytes** [comment]
> > Would definitely like to look into websockets. Hope it's possible.
> 
> Actually, I have already implemented WebSocket in my fork, you may want to do a pull and merge, but it will be tough as there are too many changes. I am currently interested and doing adding songs by speaking the song title, using OpenAI's Whisper model to do speech-to-text.

@xuancong84 I tried your fork on a raspberry pi 4 4gb. It is not running very well

**Comment by xuancong84** [comment]
> @xuancong84 I tried your fork on a raspberry pi 4 4gb. It is not running very well

My fork has been deprecated. Use my new repository at https://github.com/xuancong84/OpenHomeKaraoke.git
My home Karaoke system is using this, so I am pretty sure it works.

**Comment by killerbytes** [comment]
Yeah, I tried that on a 4GB. There's popping sound when volume normalization is enabled I think

**Comment by xuancong84** [comment]
> Yeah, I tried that on a 4GB. There's popping sound when volume normalization is enabled I think

Volume normalization only sets a fixed volume (based on the entire MP3 volume), the popping sound is due to CPU too slow while calculating the volume for a song. Try using better/faster Raspberry Pi.

**Comment by Apapora** [comment]
I reviewed your code and it still seems like it's using HTTP GET requests set over a repeating interval rather than using websockets. 
In splash.html:
`  $(function () {
    getNowPlaying(); //Calls GET request
    setInterval(getNowPlaying, 1000);
  });`

Am I missing something?

**Comment by vicwomg** [comment]
Please keep discussion to pikaraoke, forks are not officially supported.

Websocket is indeed nice for realtime applications and more performant. But there are some unexpected benefits to doing http polling, the most significant of which being we can run pikaraoke on a single port. This makes remote server and docker deploys (which are becoming more and more common) very simple. 

The polling payload is under 500 bytes. I have never experienced a performance issue.

**Comment by xuancong84** [comment]
@Apapora I am conservative in deleting codes, so what appears in my code might not get used or executed at all. For example, my code keeps both websocket version and legacy version (the legacy version won't get invoked unless you explicitly type `/home.html` instead of `/` in URL.

@vicwomg If you are only using it by yourself, you won't experience the speed issue. However, when I invited >5 friends to my home, and they are all at the home page, some did experience a bit of latency response issue (might also be due to my home wifi signal?). That is why I switched to websocket. Imagine if it is a bigger party of >10 friends, then the problem can be even more pronounced.

**Comment by vicwomg** [comment]
Websocket has been implemented in 1.10.0
